From b876c0d2b404b48a409ecea04191b9400b530887 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 22 Mar 2021 11:42:59 +0000
Subject: GCC: fix type confusion in pcscan

GCC gets confused if an alias is declared with name of another
type, e.g. "using IdType = IdType<Context>". This can be avoided
by adding explicit namespace.

Bug: 819294
Change-Id: Id4dc8a88fd63c5741a8fcb19bbdb1b7ae6d3336d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2773272
Commit-Queue: Anton Bikineev <bikineev@chromium.org>
Reviewed-by: Michael Lippautz <mlippautz@chromium.org>
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Reviewed-by: Anton Bikineev <bikineev@chromium.org>
Cr-Commit-Position: refs/heads/master@{#865110}
---
 base/allocator/partition_allocator/starscan/pcscan.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/allocator/partition_allocator/starscan/pcscan.cc b/base/allocator/partition_allocator/starscan/pcscan.cc
index f5fab3633081..d03367f40d82 100644
--- a/base/allocator/partition_allocator/starscan/pcscan.cc
+++ b/base/allocator/partition_allocator/starscan/pcscan.cc
@@ -280,7 +280,7 @@ class StatsCollector final {
   template <Context context>
   class DeferredTraceEventMap final {
    public:
-    using IdType = IdType<context>;
+    using IdType = StatsCollector::IdType<context>;
     using UnderlyingMap = MetadataHashMap<
         PlatformThreadId,
         std::array<DeferredTraceEvent, static_cast<size_t>(IdType::kNumIds)>>;
-- 
2.25.1

